homeassistant/components/elmax/alarm_control_panel.py

```diff
@@ -0,0 +1,111 @@
+"""Elmax sensor platform."""
+from __future__ import annotations
+
+from elmax_api.model.alarm_status import AlarmArmStatus, AlarmStatus
+from elmax_api.model.command import AreaCommand
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityFeature,
+    CodeFormat,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_ALARM_ARMED_AWAY, STATE_ALARM_DISARMED
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import InvalidStateError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax area platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
```

Comment:
Trying to understand the need for this. How would come across devices that you've already configured after the integration is set up?

Reply:
This is a way of handling the configuration changes of Elmax devices without requesting a full restart of HA. In particular users can enable/disable devices via the Elmax web interface. When they do so, the coordinator will receive a state object which contains the new entities: by intercepting that and fetching the new devices, we can directly add them in HA.

Does this clarify? Let me know if it doesn't.

Reply:
Ah, got it! 👍🏻